### PR TITLE
partially fix #37984: support round and floor for AbstractFloat

### LINKED ISSUE #37984: no trunc/round/floor/ceil(T, x) for Real T
Not sure if this was intended, but there is no conversion allowed to Float64 (Float32 and maybe other Real) after applying `trunc()` with the type explicitly specified in the call arguments.

```
julia> trunc(Float64, 1.5)
ERROR: MethodError: no method matching trunc(::Type{Float64}, ::Float64)
Closest candidates are:
  trunc(::Type{UInt128}, ::Float64) at float.jl:678
  trunc(::Type{Int8}, ::Float64) at float.jl:678
  trunc(::Type{UInt32}, ::Float64) at float.jl:678
  ...
Stacktrace:
 [1] top-level scope at REPL[72]:1
```

The same goes for `round`, `ceil`, and `floor` and I think this behavior is inconsistent considering other similar uses.

```
julia> trunc(1.5)
1.0
```

```
julia> trunc(Float64, 3//2)
1.0
```

### PR DESCRIPTION
**Pull request by mivanovska007**
adding floor and ceil for floating types

### CODE REVIEW

File: test/rounding.jl [hunk 1]
```diff
@@ -351,3 +351,10 @@ end
         Base.Rounding.setrounding_raw(T, Base.Rounding.to_fenv(old))
     end
 end
+
+@testset "rounding to AbstractFloat" begin
+    for T in [Float16, Float32, Float64]
+        @test floor(T, 2.3) <= 2.3
```
Comment (johnnychen94) [inline test/rounding.jl:357]: Testing this is probably not enough, we also need to test that this doesn't return `floor(2.3) == 2.0`.
It's also worth adding a comment about the fact that `Float16(2.3) > 2.3` and `Float32(2.3) < 2.3`.

**Review by johnnychen94 — COMMENTED** [review]
@mivanovska007 can you change the title of this PR to a more meaningful one, e.g., "support round and floor for AbstractFloat"? Otherwise, people need to click the URL first before understanding what this PR does.

I'm starting to wonder what's the benefit of this `floor` over `prefloat` and `ceil` over ` nextfloat`.

How about `round` then, still not implemented, or just something equivalent to `convert(T, x)`?

I'm not sure if people will get confused at first glance when they see `floor(Float16, 2.3) != 2.0`. But still, docstrings and examples are needed to clarify the behavior

**Comment by mivanovska007** [comment]
>I'm not sure if people will get confused at first glance when they see floor(Float16, 2.3) != 2.0. But still, docstrings and examples are needed to clarify the behavior

According to the doc,
>floor(x) returns the nearest integral value of the same type as x that is less than or equal to x.
floor(T, x) converts the result to type T, throwing an InexactError if the value is not representable.

so I would expect
```julia
convert(Float64, 2.3)
convert(2.3)
```
to return the same thing, so maybe the correct thing is to not support `floor(::AbstractFloat, x)` to avoid confusion.

**Comment by johnnychen94** [comment]
> so maybe the correct thing is to not support floor(::AbstractFloat, x) to avoid confusion.

Since we both agree on it, let's say, close this PR and the issue?

**Pull request closed without merging**